anesthetic/weighted_pandas.py

```diff
@@ -617,7 +617,28 @@ def compress(self, ncompress=True, axis=0):
             df = df.set_axis(self._get_axis(1-axis), axis=1-axis, copy=False)
             return df
         else:
-            return self
+            # For unweighted data, if ncompress is an integer > 0,
+            # randomly subsample
+            if (isinstance(ncompress, int) and
+                    not isinstance(ncompress, bool) and ncompress > 0):
```

Comment:
We can drop the `not isinstance(ncompress, bool)`. That is already excluded by `isinstance(ncompress, int)`.

Also, this `if` could be combined with the preceding `else` to an `elif`, that might help reduce line wrapping.